pkg/config/integration_config.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/google/exposure-notifications-server/pkg/keys"
+	"github.com/google/exposure-notifications-server/pkg/observability"
+	"github.com/google/exposure-notifications-verification-server/pkg/cache"
+	"github.com/google/exposure-notifications-verification-server/pkg/database"
+	"github.com/google/exposure-notifications-verification-server/pkg/ratelimit"
+)
+
+const (
+	VerificationTokenDuration = time.Second * 2
+	APIKeyCacheDuration       = time.Second * 2
+
+	APISrvPort   = "8080"
+	AdminSrvPort = "8081"
+)
+
+// IntegrationTestConfig represents configurations to run server integration tests.
+type IntegrationTestConfig struct {
+	Observability *observability.Config
+	DBConfig      *database.Config
+
+	APISrvConfig      APIServerConfig
+	AdminAPISrvConfig AdminAPIServerConfig
+}
+
+func NewIntegrationTestConfig(ctx context.Context, tb testing.TB) (*IntegrationTestConfig, *database.Database) {
+	db, dbConfig := database.NewTestDatabaseWithConfig(tb)
+	obConfig := &observability.Config{
+		ExporterType: "NOOP",
+	}
+
+	cacheConfig := cache.Config{
+		Type: "IN_MEMORY",
+	}
+
+	rlConfig := ratelimit.Config{
+		Type: "NOOP",
+	}
+
+	tmpdir, err := ioutil.TempDir("", "")
+	if err != nil {
+		tb.Fatal(err)
+	}
+	keyConfig := keys.Config{
+		KeyManagerType: "FILESYSTEM",
+		FilesystemRoot: tmpdir,
+	}
+
+	kms, err := keys.KeyManagerFor(ctx, &keyConfig)
+	if err != nil {
+		tb.Fatal(err)
+	}
+	tb.Cleanup(func() {
+		if err := os.RemoveAll(tmpdir); err != nil {
```

Comment:
Shouldn't this be best effort and not failing test if it failed?

Reply:
I think this matches what we do elsewhere.